src/util/generate_server_certificate.c

```diff
@@ -0,0 +1,119 @@
+/******************************************************************************
+ *
+ * Copyright 2018 Xaptum, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License
+ *
+ *****************************************************************************/
+#include <xtt.h>
+#include <string.h>
+#include <time.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+int generate_server_certificate(const char* root_cert_file, const char* root_privatekey_file, const char* server_id_file, const char* expiry_in,
+                            const char* server_certificate_filename, const char* server_privatekey_file)
+{
+    xtt_ecdsap256_pub_key root_pub;
+    xtt_ecdsap256_priv_key root_priv;
+    xtt_certificate_root_id root_id;
+    xtt_identity_type server_id;
+    xtt_root_certificate root_certificate;
+
+    //Read expiry, and check validity, if none given add one year from today
+    xtt_certificate_expiry expiry;
+    if(expiry_in != NULL){
+        memcpy(expiry.data, expiry_in, 8);
+    } else {
+        time_t today = time(NULL);
```

Comment:
Would you include the comments I put in my example code, in the code you included in the file (here, and from the comment above)? Just to make sure anyone reading the code is clear on, e.g. why we add `1` to the `tm_mon`.

Reply:
Let's keep the `xtt_create_expiry` function out of the public API, since it's just used internally.

So, remove it from the header `generate_server_certificate.h`.

Also, I personally like making such internal functions (ones that are used only within a single `.c` file) `static`. That means making the definition of `xtt_create_expiry` at the top of the `.c` file look like:
```
static
int xtt_create_expiry(...
```

This keeps the scope of this internal function actually local to this file (by limiting the linkage), so that makes sure no one else can accidentally use this function. It also has minor performance implications, related to shared library creation, but that's a whole other story...